ts/Core/Templating.ts

```diff
@@ -173,10 +174,13 @@ function dateFormat(
  * @param {Highcharts.Chart} [chart]
  *        A `Chart` instance used to get numberFormatter and time.
  *
+ * @param {Highcharts.Time} [time]
+ *        A `Time` instance used to get dateFormat.
+ *
  * @return {string}
  *         The formatted string.
  */
-function format(str = '', ctx: any, chart?: Chart): string {
+function format(str = '', ctx: any, chart?: Chart, time?: Time): string {
```

Comment:
Since both Chart and DataGrid have the `time` property, can we avoid adding the fourth parameter? Then the third parameter would be `Chart|DataGrid`.

Line 193 would require that we have a `dataGrid.options.lang` property, that I assume we already do. 

Line 195 - either we should have a `dataGrid.numberFormatter` format, or if we don't we can do `(chart as Chart)?.numberFormatter || numberFormat;`

Reply:
You are right this will work but the only problem I have with this approach is that all DataGrid files are excluded in the `tsconfig` and we probably don't want to include them in the core build right?

Reply:
Probably. But maybe we can get away with a simplified interface that both Chart and DataGrid are compatible with? Something like...

```
interface TemplatingOwnerOptions {
   lang: LangOptions;
}
interface TemplatingOwner {
    options: TemplatingOwnerOptions;
    time?: Time;
    numberFormatter?: Function.
}
```